packages/core/src/components/icon/icon.tsx

```diff
@@ -101,7 +101,7 @@ export interface DefaultIconProps extends IntentProps, Props, DefaultSVGIconProp
  * @see https://stackoverflow.com/a/73795494/7406866
  */
 export interface IconComponent extends React.FC<IconProps<Element>> {
-    <T extends Element = Element>(props: IconProps<T>): React.ReactElement | null;
+    <T extends Element = Element>(props: IconProps<T>): ReturnType<React.FC<IconProps<Element>>> | null;
```

Comment:
Might be worth a comment that this is written in a certain way to guarantee React 16 + 18 compatability!

Reply:
Done! bf41f431b07d31f469b88d0a2933097ff124dabd